TF_Service_dotNet/TouchFree Service/InteractionManager.cs

```diff
@@ -27,23 +30,68 @@ public InteractionManager(UpdateBehaviour _updateBehaviour, HandManager _handMan
             hoverAndHold = new HoverAndHoldInteraction(handManager);
             touchPlane = new TouchPlanePushInteraction(handManager);
 
-            // This will need to be swapped with a system to add/remove these listeners per the "active" interaction
+            ConfigManager.OnInteractionConfigUpdated += OnInteractionSettingsUpdated;
+
+            OnInteractionSettingsUpdated(ConfigManager.InteractionConfig);
+        }
+
+        protected void OnInteractionSettingsUpdated(InteractionConfig _config)
+        {
+            switch(lastInteraction)
+            {
+                case InteractionType.TOUCHPLANE:
+                    updateBehaviour.OnUpdate -= touchPlane.Update;
+                    touchPlane.HandleInputAction -= connectionManager.SendInputActionToWebsocket;
+                    touchPlane.Disable();
```

Comment:
Could these switch statements make use of generic EnableInteraction and DisableInteraction functions?

like this:

EnableInteraction(InteractionModule _module)
{
	updateBehaviour.OnUpdate += _module.Update;
	_module.HandleInputAction += connectionManager.SendInputActionToWebsocket;
	_module.Enable();
}